### PR TITLE
Version check medic/medic-webapp#2931 medic/medic-webapp#2854

### PR DESCRIPTION
**Pull request by SCdF**
Wow that diffs badly doesn't it. I only wrapped and added the first check

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -650,34 +650,56 @@ proxyForAuditing.on('error', function(err, req, res) {
...
-ddocExtraction.run(function(err) {
+var couchdbCheck = function(callback) {
+  db.request({}, function(err, body) {
+    if (err) {
+      callback(err);
+    }
+
+    var semvers = body.version && body.version.match(/(\d+)\.\d+\.\d+/);
+    if (!semvers || semvers[1] !== '2') {
+      callback('Only CouchDB 2.x is supported, v' + body.version + ' detected.');
```
Comment (garethbowen) [inline server.js]: Make this `callback(new Error('Only...'));` so we get a stacktrace.

File: server.js [hunk 2]
```diff
@@ -650,34 +650,56 @@ proxyForAuditing.on('error', function(err, req, res) {
...
+    }
+
+    callback();
+  });
+};
+
+couchdbCheck(function(err) {
```
Comment (garethbowen) [inline server.js]: This is too deeply nested now. It was too deeply nested before, but now it's just crazy! Maybe use async.series or something.

Comment (SCdF) [inline server.js]: It's funny, I actually wrote it with promises, then realised "whoops, wrong standard" and changed it to this. I will try to clean it up.

File: server.js (hunk 1 above)
Comment (SCdF) [inline server.js]: rgr that

**Pull request merged**